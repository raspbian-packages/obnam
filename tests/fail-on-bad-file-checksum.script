#!/bin/sh
# Copyright 2011  Lars Wirzenius
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

set -e

$SRCDIR/tests/backup

# Remove the chunk checksum list, and then modify the chunks,
# so that we can do a restore without triggering "bad chunk checksum"
# errors. We only want to trigger the whole-file checksum error.

rm -rf "$DATADIR/repo/chunklist/nodes"
rm -rf "$DATADIR/repo/chunklist/refcounts"
rm -rf "$DATADIR/repo/chunklist/metadata"

find "$DATADIR/repo/chunks" -type f |
while read filename
do
    tr '\0-\377' '\200-\377\0-\177' < "$filename" > "$filename.new"
    mv "$filename.new" "$filename"
done

# Restore.
$SRCDIR/tests/restore

