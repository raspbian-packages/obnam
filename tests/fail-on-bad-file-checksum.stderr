ERROR: There were errors when restoring
